FreeRTOS/Demo/RX700_RX72N_EnvisionKit_GCC_e2studio/src/frtos_startup/freertos_start.c

```diff
@@ -0,0 +1,483 @@
+* ANY REASON RELATED TO THIS SOFTWARE, EVEN IF RENESAS OR ITS AFFILIATES HAVE
+* BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
+* Renesas reserves the right, without notice, to make changes to this software
+* and to discontinue the availability of this software. By using this software,
+* you agree to the additional terms and conditions found by accessing the 
+* following link:
+* http://www.renesas.com/disclaimer
+* Copyright (C) 2018 Renesas Electronics Corporation. All rights reserved.    
+*******************************************************************************/
+/*******************************************************************************
+* File Name    : freertos_start.c
+* Version      : 1.0
+* Description  : Contains FreeRTOS user-defined functions.
+******************************************************************************/
+/*****************************************************************************
+* History : DD.MM.YYYY Version  Description
+*         : 30.06.2016 1.00     First Release
+******************************************************************************/
+
+/******************************************************************************
+Includes   <System Includes> , "Project Includes"
+******************************************************************************/
+#include "platform.h"
+#include "freertos_start.h"
+#include "demo_main.h"
+#include "demo_specific_io.h"
+
+#if (BSP_CFG_RTOS_USED == 1)
+
+#if (RTOS_USB_SUPPORT == 1)
+#include "r_usb_basic_if.h"
+#include "r_usb_cstd_rtos.h"
+#endif
+
+/******************************************************************************
+Macro definitions
+******************************************************************************/
+
+/******************************************************************************
+Typedef definitions
+******************************************************************************/
+
+/******************************************************************************
+Private global variables and functions
+******************************************************************************/
+
+/************* semaphore initialization *****************/
+
+/******************************************************************************
+Exported global variables (to be accessed by other files)
+******************************************************************************/
+
+/******************************************************************************
+Exported global functions (to be accessed by other files)
+******************************************************************************/
+
+/* FreeRTOS's system timer. */
+void vApplicationSetupTimerInterrupt(void);
+
+/* Hook functions used by FreeRTOS. */
+void vAssertCalled(void);
+void vApplicationIdleHook(void);
+void vApplicationTickHook(void);
+void vApplicationMallocFailedHook(void);
+void vApplicationStackOverflowHook(TaskHandle_t xTask, signed char *pcTaskName);
```

Comment:
the pcTaskName needs to be a char* to match the definition in task.c.
That extern definition in task.c will shortly move to a prototype in task.h so this will generate a warning as the prototypes will not match.

Reply:
Thank you for your checking.

I will merge recent FreeRTOS-kernel's (and FreeRTOS's) commits and fix the warning.